### PR TITLE
when removing from a collection in reference mode, clear the entity f…

### PR DESCRIPTION
**Pull request by galganif**
…rom the backing store, to avoid leaving potentially sensitive data around

### CODE REVIEW

File: javatests/arcs/core/storage/ReferenceModeStoreTest.kt [hunk 1]
```diff
@@ -254,6 +254,34 @@ class ReferenceModeStoreTest {
...
+        val deleteOp = RefModeStoreOp.SetRemove(actor, VersionMap(actor to 1), bob)
+        assertThat(
+            activeStore.onProxyMessage(ProxyMessage.Operations(listOf(deleteOp), id = 1))
+        ).isTrue()
+
+        // Check the backing store Bob has been cleared.
+        val storedBob2 = activeStore.backingStore.getLocalData("an-id") as CrdtEntity.Data
+        assertThat(storedBob2.toRawEntity("an-id")).isEqualTo(createEmptyPersonEntity("an-id"))
+    }
+
```
Comment (jasonwyatt) [inline javatests/arcs/core/storage/ReferenceModeStoreTest.kt]: This looks great, can you add a similar test to the variants of the ReferenceModeStoreIntegrationTest?

Comment (galganif) [reply javatests/arcs/core/storage/ReferenceModeStoreTest.kt]: Yes you were right, added tests with the database impl and had to fix an issue with null values. I am not sure there are equivalent tests with ramdisks?

File: java/arcs/android/storage/database/DatabaseImpl.kt [hunk 2]
```diff
@@ -389,45 +389,55 @@ class DatabaseImpl(
...
-                SQLiteDatabase.CONFLICT_REPLACE
-            )
+                counters?.increment(DatabaseCounters.UPDATE_ENTITY_FIELD_VALUE)
+                insertWithOnConflict(
+                    TABLE_FIELD_VALUES,
+                    null,
+                    content,
+                    SQLiteDatabase.CONFLICT_REPLACE
+                )
         }
```
Comment (jasonwyatt) [inline java/arcs/android/storage/database/DatabaseImpl.kt]: I think this now needs to be bumped one indent level deeper?

File: javatests/arcs/core/storage/ReferenceModeStoreDatabaseIntegrationTest.kt [hunk 3]
```diff
@@ -222,6 +222,45 @@ class ReferenceModeStoreDatabaseIntegrationTest {
...
+        val bob = createPersonEntity("an-id", "bob", 42)
+
+        // Add Bob to collection.
+        val addOp = RefModeStoreOp.SetAdd(actor, VersionMap(actor to 1), bob)
+        assertThat(
+            activeStore.onProxyMessage(ProxyMessage.Operations(listOf(addOp), id = 1))
+        ).isTrue()
+        // Bob was added to the backing store.
+        val storedBob = activeStore.backingStore.getLocalData("an-id") as CrdtEntity.Data
+        assertThat(storedBob.toRawEntity("an-id")).isEqualTo(bob)
```
Comment (jasonwyatt) [inline javatests/arcs/core/storage/ReferenceModeStoreDatabaseIntegrationTest.kt]: Nit: the ordering for assertThat is meant to be `assertThat(actual).isEqualTo(expected)`, which - in this case, makes me think `bob` should be within the `assertThat` call, and storedBob would be in the `isEqualTo` call.

File: javatests/arcs/core/storage/ReferenceModeStoreTest.kt [hunk 4]
```diff
@@ -254,6 +254,34 @@ class ReferenceModeStoreTest {
...
+        val bob = createPersonEntity("an-id", "bob", 42)
+
+        // Add Bob to collection.
+        val addOp = RefModeStoreOp.SetAdd(actor, VersionMap(actor to 1), bob)
+        assertThat(
+            activeStore.onProxyMessage(ProxyMessage.Operations(listOf(addOp), id = 1))
+        ).isTrue()
+        // Bob was added to the backing store.
+        val storedBob = activeStore.backingStore.getLocalData("an-id") as CrdtEntity.Data
+        assertThat(storedBob.toRawEntity("an-id")).isEqualTo(bob)
```
Comment (jasonwyatt) [inline javatests/arcs/core/storage/ReferenceModeStoreTest.kt]: same actual vs. expected order nit here

**Pull request merged**